### PR TITLE
export file adjustments

### PR DESCRIPTION
**Pull request by mikehelmick**
## Proposed Changes

* when splitting an export window into multi size 1 batches, adjust the end timestamp to avoid hash collisions on Android device matching.

**Release Note**

```release-note
Update batch num/size and timestamp handling
```

### CODE REVIEW

File: internal/export/exportfile.go [hunk 1]
```diff
@@ -176,6 +176,8 @@ func assignReportType(reportType *string, pbek *export.TemporaryExposureKey) {
 	}
 }
 
+// The batch num and batch size are always set to 1/1.
```
Comment (flagxor) [inline internal/export/exportfile.go:179]: Maybe clarify you mean in the output, as opposed to the function assumes the inputs are 1/1?

File: internal/export/worker.go [hunk 2]
```diff
@@ -305,15 +305,15 @@ func (s *Server) exportBatch(ctx context.Context, eb *model.ExportBatch, emitInd
...
 		objectName, err := s.createFile(ctx,
 			createFileInfo{
 				exposures:        group.exposures,
 				revisedExposures: group.revised,
 				exportBatch:      eb,
 				signatureInfos:   sigInfos,
-				batchNum:         1,
-				batchSize:        1,
-				groupNum:         1,
+				batchNum:         i + 1,
```
Comment (flagxor) [inline internal/export/worker.go:315]: Why does this come back in?

File: internal/export/exportfile.go [hunk 3]
```diff
@@ -209,12 +211,16 @@ func marshalContents(eb *model.ExportBatch, exposures, revisedExposures []*publi
...
+		offset = int64(batchNum)
+	}
 	pbeke := export.TemporaryExposureKeyExport{
 		StartTimestamp: proto.Uint64(uint64(eb.StartTimestamp.Unix())),
-		EndTimestamp:   proto.Uint64(uint64(eb.EndTimestamp.Unix())),
+		EndTimestamp:   proto.Uint64(uint64(eb.EndTimestamp.Unix() + offset)),
 		Region:         proto.String(eb.OutputRegion),
-		BatchNum:       proto.Int32(batchNum),
-		BatchSize:      proto.Int32(batchSize),
+		BatchNum:       proto.Int32(1),
```
Comment (whaught) [inline internal/export/exportfile.go:222]: I'm confused about why we're hardcoding BatchNum 1 and size 1

File: internal/export/worker.go [hunk 4]
```diff
@@ -365,7 +364,7 @@ func (s *Server) createFile(ctx context.Context, cfi createFileInfo) (string, er
 		return "", fmt.Errorf("marshaling export file: %w", err)
 	}
 
-	objectName := exportFilename(cfi.exportBatch, cfi.groupNum, pbHexSHA)
+	objectName := exportFilename(cfi.exportBatch, cfi.batchNum, pbHexSHA)
```
Comment (flagxor) [inline internal/export/worker.go:367]: Why does the groupNum go away here but not in the filename below?

**Comment by mikehelmick** [comment]
addressed all comments by write comments int he code

**Review by flagxor — APPROVED** [review]
lgtm

File: internal/export/exportfile.go [hunk 5]
```diff
@@ -176,6 +176,17 @@ func assignReportType(reportType *string, pbek *export.TemporaryExposureKey) {
...
 }
 
+// The batch num and batch size are always set to 1 / 1 when actually generating a file.
+// This is to avoid a device side issue where all batches must be passed in toegher.
+// By making all batches size one, we make the unit of atomicity a single export file, avoiding
+// the need for clients to do any kind of batching before passing to the OS.
+//
+// If this is part of a larger batch, the end timestamps are adjusted. Android hashes
+// the start/end/batchNum to de-duplicate. If there are X files that have the same timing
+// metadata, then only the first would get processed. We compensate her by bumping the end
```
Comment (whaught) [inline internal/export/exportfile.go:186]:
```suggestion
// metadata, then only the first would get processed. We compensate here by bumping the end
```

File: internal/export/exportfile.go [hunk 6]
```diff
@@ -209,12 +220,16 @@ func marshalContents(eb *model.ExportBatch, exposures, revisedExposures []*publi
...
+	}
 	pbeke := export.TemporaryExposureKeyExport{
 		StartTimestamp: proto.Uint64(uint64(eb.StartTimestamp.Unix())),
-		EndTimestamp:   proto.Uint64(uint64(eb.EndTimestamp.Unix())),
+		EndTimestamp:   proto.Uint64(uint64(eb.EndTimestamp.Unix() + offset)),
 		Region:         proto.String(eb.OutputRegion),
-		BatchNum:       proto.Int32(batchNum),
-		BatchSize:      proto.Int32(batchSize),
+		BatchNum:       proto.Int32(1), // all batches are now size 1 (singele file)
+		BatchSize:      proto.Int32(1), // so it's alwys 1 of 1.
```
Comment (whaught) [inline internal/export/exportfile.go:232]:
```suggestion
		BatchSize:      proto.Int32(1), // so it's always 1 of 1.
```

File: internal/export/exportfile.go [hunk 7]
```diff
@@ -209,12 +220,16 @@ func marshalContents(eb *model.ExportBatch, exposures, revisedExposures []*publi
...
+		offset = int64(batchNum)
+	}
 	pbeke := export.TemporaryExposureKeyExport{
 		StartTimestamp: proto.Uint64(uint64(eb.StartTimestamp.Unix())),
-		EndTimestamp:   proto.Uint64(uint64(eb.EndTimestamp.Unix())),
+		EndTimestamp:   proto.Uint64(uint64(eb.EndTimestamp.Unix() + offset)),
 		Region:         proto.String(eb.OutputRegion),
-		BatchNum:       proto.Int32(batchNum),
-		BatchSize:      proto.Int32(batchSize),
+		BatchNum:       proto.Int32(1), // all batches are now size 1 (singele file)
```
Comment (whaught) [inline internal/export/exportfile.go:231]:
```suggestion
		BatchNum:       proto.Int32(1), // all batches are now size 1 (single file)
```

Comment (flagxor) [inline internal/export/exportfile.go:231]: typo on singele

**Comment by mikehelmick** [comment]
@whaught 
merged in your suggestions, please lgtm again

**Review by flagxor — APPROVED** [review]

**Review by whaught — APPROVED** [review]

**Pull request merged**